File: java/src/main/java/com/android/widget/FloatWindow/NeedPermission/ScreenType.java
package com.android.widget.FloatWindow.NeedPermission;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Created by xuzhb on 2021/3/8
 * Desc:选择以屏幕宽度或屏幕高度为基准
 */
public class ScreenType {
    public static final int width = 0;
    public static final int height = 1;

    @IntDef({width, height})
    @Retention(RetentionPolicy.SOURCE)
    @interface screenType {
    }
}

File: kotlin/src/main/java/com/android/widget/FloatWindow/NeedPermission/ScreenType.kt
package com.android.widget.FloatWindow.NeedPermission

import androidx.annotation.IntDef

/**
 * Created by xuzhb on 2021/3/29
 * Desc:选择以屏幕宽度或屏幕高度为基准
 */
object ScreenType {
    const val width = 0
    const val height = 1

    @IntDef(width, height)
    @kotlin.annotation.Retention(AnnotationRetention.SOURCE)
    internal annotation class screenType
}